index/scorch/scorch.go

```diff
@@ -466,33 +466,59 @@ func (s *Scorch) Stats() json.Marshaler {
 	return &s.stats
 }
 
-func (s *Scorch) diskFileStats() (uint64, uint64) {
-	var numFilesOnDisk, numBytesUsedDisk uint64
+func (s *Scorch) diskFileStats(rootSegmentPaths map[string]struct{}) (uint64,
+	uint64, uint64) {
+	var numFilesOnDisk, numBytesUsedDisk, numBytesOnDiskByRoot uint64
 	if s.path != "" {
 		finfos, err := ioutil.ReadDir(s.path)
 		if err == nil {
 			for _, finfo := range finfos {
 				if !finfo.IsDir() {
 					numBytesUsedDisk += uint64(finfo.Size())
 					numFilesOnDisk++
+					if rootSegmentPaths != nil {
+						fname := s.path + string(os.PathSeparator) + finfo.Name()
+						if _, fileAtRoot := rootSegmentPaths[fname]; fileAtRoot {
+							numBytesOnDiskByRoot += uint64(finfo.Size())
+						}
+					}
 				}
 			}
 		}
 	}
-	return numFilesOnDisk, numBytesUsedDisk
+	// if no root files path given, then consider all disk files.
+	if rootSegmentPaths == nil {
+		numBytesOnDiskByRoot = numBytesUsedDisk
+	}
+
+	return numFilesOnDisk, numBytesUsedDisk, numBytesOnDiskByRoot
+}
+
+func (s *Scorch) rootDiskSegmentsPaths() map[string]struct{} {
+	rv := make(map[string]struct{}, len(s.root.segment))
+	for _, segmentSnapshot := range s.root.segment {
+		switch seg := segmentSnapshot.segment.(type) {
+		case *zap.Segment:
+			rv[seg.Path()] = struct{}{}
+		}
+	}
+	return rv
 }
 
 func (s *Scorch) StatsMap() map[string]interface{} {
 	m := s.stats.ToMap()
 
-	numFilesOnDisk, numBytesUsedDisk := s.diskFileStats()
+	s.rootLock.RLock()
+	rootSegPaths := s.rootDiskSegmentsPaths()
+	m["CurFilesIneligibleForRemoval"] = uint64(len(s.ineligibleForRemoval))
+	s.rootLock.RUnlock()
+
+	numFilesOnDisk, numBytesUsedDisk, numBytesOnDiskByRoot := s.diskFileStats(rootSegPaths)
 
 	m["CurOnDiskBytes"] = numBytesUsedDisk
 	m["CurOnDiskFiles"] = numFilesOnDisk
+	m["CurOnDiskBytesByRoot"] = numBytesOnDiskByRoot
```

Comment:
Can you add a description for each of these new stats to the commit message?  Just like a sentence is enough.  The problem is that the stat name uses "root" which we use inside of scorch to mean a particular thing, but that isn't obvious to people outside of the scorch internals.

Reply:
CurOnDiskBytesByIndex or CurOnDiskIndexBytes ?  
Currently we don't have a stat which explicitly gives the index size on disk.  CurOnDiskBytes computes the whole files size in the index directory which might include the older snapshots as well.
Exact index disk size helps to suggest a resident ratio/ram quota for the index to the customer - is the context for this change,

Reply:
So, my request was for you to add it to the commit message, as I think that people are more likely to consult that if they have questions about it.

Second, I disagree with the statement 'we don't have a stat which explicitly gives the index size on disk'.  If you have chosen to keep more than 1 snapshot, or you have queries using older snapshots, then your index does contain multiple snapshots.  So, the size of your index is the size of all of them.

That doesn't mean there isn't value in knowing the size of just the current root.  We just disagree on the definitions.

Reply:
do you have a better naming suggestion as we disagree here on the definitions?

Reply:
ok, let me go back to the older CurOnDiskBytesByRoot name itself your request was about just changing the commit msg.

Reply:
i sense that you don't agree about something, but i don't ow what that is

Reply:
i just misread your original comment about renaming the stat from root -> index, but then on a second read I realised that its more about the commit description than about the exact naming stuff..